### PR TITLE
Crdt demo

### PR DESCRIPTION
**Pull request by hahn-kev**
features

- upload project to server
- set dev mode in browser from backend
- filter changes pushed by current filter
- add missing entries that come from pushed changes

### CODE REVIEW

**Review by myieye — CHANGES_REQUESTED** [review]
This is just a partial review sadly.
I'll try to finish tomorrow.
I'm posting it to make my progress official 😉.

File: backend/FwLite/LocalWebApp/Routes/AuthRoutes.cs [hunk 1]
```diff
@@ -19,13 +19,13 @@ public static IEndpointConventionBuilder MapAuthRoutes(this WebApplication app)
...
                     !string.IsNullOrEmpty(currentName),
-                    currentName);
+                    currentName, s.Authority.Authority);
             });
         });
         group.MapGet("/login/{server}",
             async (AuthHelpersFactory factory, string server, IOptions<AuthConfig> options) =>
             {
-                var result = await factory.GetHelper(options.Value.GetServer(server)).SignIn();
+                var result = await factory.GetHelper(options.Value.GetServerByAuthority(server)).SignIn();
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/LocalWebApp/Routes/AuthRoutes.cs:28]: good catch, it was a left over from the previous name

File: backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs [hunk 2]
```diff
@@ -21,11 +23,51 @@ public override async Task OnConnectedAsync()
     {
         await Groups.AddToGroupAsync(Context.ConnectionId, ProjectGroup(projectContext.Project.Name));
         await syncService.ExecuteSync();
-        changeEventBus.SetupGlobalSignalRSubscription();
+        IDisposable[] cleanup =
+        [
+            //todo this results in a memory leak, due to holding on to the hub instance, it will be disposed even if the context items are not.
+            changeEventBus.ListenForEntryChanges(projectContext.Project.Name, Context.ConnectionId)
+        ];
+        Context.Items["clanup"] = cleanup;
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs:31]: nice catch, I refactored this a bit

File: backend/FwLite/FwLiteDesktop/MainPage.xaml.cs [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
+        _logger = logger;
+        _environment = environment;
         InitializeComponent();
         options.OnChange(o =>
         {
             var url = o.Url;
             webView.Dispatcher.Dispatch(() => webView.Source = url);
             logger.LogInformation("Url updated: {Url}", url);
         });
+        webView.IsVisible = false;
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/FwLiteDesktop/MainPage.xaml.cs:23]: yeah I think it was left over from when I was troubleshooting something else

File: backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs [hunk 4]
```diff
@@ -21,11 +23,51 @@ public override async Task OnConnectedAsync()
     {
         await Groups.AddToGroupAsync(Context.ConnectionId, ProjectGroup(projectContext.Project.Name));
         await syncService.ExecuteSync();
-        changeEventBus.SetupGlobalSignalRSubscription();
+        IDisposable[] cleanup =
+        [
+            //todo this results in a memory leak, due to holding on to the hub instance, it will be disposed even if the context items are not.
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs:28]: I can't remember why I put that comment there, and I don't think it's valid. I also actually debugged the application and it doesn't look like there's a leak there, so I'm not worried.

File: backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs [hunk 5]
```diff
@@ -21,11 +23,51 @@ public override async Task OnConnectedAsync()
...
+            Filtering.CompiledFilter(null, options?.Exemplar?.WritingSystem ?? "default", options?.Exemplar?.Value);
+        return base.GetEntries(options);
+    }
+
+    public override IAsyncEnumerable<Entry> SearchEntries(string query, QueryOptions? options = null)
+    {
+        CurrentFilter = Filtering.CompiledFilter(query,
+            options?.Exemplar?.WritingSystem ?? "default",
+            options?.Exemplar?.Value);
+        return base.SearchEntries(query, options);
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/LocalWebApp/Hubs/CrdtMiniLcmApiHub.cs:73]: good thinking, I've written some tests for it, it's not exhaustive but it should get the job done

**Review by myieye — APPROVED** [review]
Mostly LGTM 😎 

I added 2 tiny commits as I found small UI bugs and such...maybe I should have put them in my own PR.

File: backend/FwLite/FwLiteDesktop/MainPage.xaml.cs [hunk 6]
```diff
@@ -39,7 +45,25 @@ public MainPage(IOptionsMonitor<LocalWebAppConfig> options, ILogger<MainPage> lo
...
+                NavigationSuccess();
+            }
         };
     }
-}
 
+    private void NavigationSuccess()
+    {
+        webView.IsVisible = true;
+        //not currently working
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/FwLite/FwLiteDesktop/MainPage.xaml.cs:58]: haha yeah it's no longer valid

**Pull request merged**